### ISSUE TITLE
Open Folder dialog not triggering directory selection

### ISSUE THREAD

**Issue by chazfg** [lapce/lapce#3217]
## Lapce Version

Nightly

## System information

Artix 6.8.7

## Describe the bug

Open Folder/File does not work. 

## Additional information

Lapce-app::window_tab includes `use floem::action::open_file` (equivalent). "open_file" no longer is located via this path in the revision specified in the Cargo.toml for Lapce-app. I tried to revert to the 1.1 release of floem in the Cargo.toml but that release of floem lacks the features needed.

![floem_action](https://github.com/lapce/lapce/assets/81254302/071d4a18-9ada-4320-9902-2ee583442b06)
![lapce_floem_rev](https://github.com/lapce/lapce/assets/81254302/be63cee6-06e8-40cd-81b8-7fd22200b0a4)
![lapce_action_open_file](https://github.com/lapce/lapce/assets/81254302/a4fd9275-76e4-4257-8030-5ebb78520e09)

**Comment by chazfg** [comment]
When opening from terminal with -w I get this when I try to open a folder

2024-04-29T02:02:10.851682Z ERROR rfd::backend::xdg_desktop_portal: pick_folder error No such file or directory (os error 2)

**Comment by chazfg** [comment]
rfd wasn't working with my current desktop portal set up. Went through, made sure that I'd had xdg-portal conf set up so that it would choose GTK for a FileChooser (since hyprland desktop portal does not have a file chooser) but there was still the above error. xdg-desktop-portal is installed. Whatever the case, rfd (and by extension the open folder dialog) work when specifying the gtk3 feature over the xdg-portal feature. I created a fork of floem to modify the way it handles the rfd dependency to test if then Lapce would work, and it did. 

Basically, I added the ability to choose between gtk3 and xdg-portal when using floem as a crate

**Comment by IamRezaMousavi** [comment]
I have this issue in KDE Plasma on Arch linux too

**Comment by introspectionism** [comment]
I have this issue in i3wm on Arch.

**Comment by indrora** [comment]
This happens on ChromeOS too

**Comment by Metgraph** [comment]
This could be related to rfd.
https://docs.rs/rfd/latest/rfd/#linux--bsd-backends

In [floem library](https://github.com/lapce/floem) manifest, rfd is set to use XDG Desktop Portal backend.
You need to install one supported XDG Desktop portal backend or set rfd to use gtk3 backend in floem manifest

**Comment by Kurumi-Tokito** [comment]
It's working for me after some testing (hope it helps)

i am using both **hyprland and gtk** portals with this script 

- portal
```
#!/usr/bin/env bash

sleep 1
killall -e xdg-desktop-portal-hyprland
killall -e xdg-desktop-portal-gtk
killall xdg-desktop-portal

/usr/lib/xdg-desktop-portal-hyprland & sleep 2
/usr/lib/xdg-desktop-portal-gtk & sleep 2
/usr/lib/xdg-desktop-portal &
```

- hyprland.conf
```
exec-once = dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once = systemctl --user import-environment WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once = $HOME/.local/share/bin/portal

env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland
env = XDG_CURRENT_DESKTOP,Hyprland
```

#### Note: There is a delay like 3~5 sec before file picker appears
